### PR TITLE
Implement Typos CI Workflow and Correct Spelling Errors Across the Project

### PR DESCRIPTION
**Pull request by liby**
## Overview

This pull request introduces a significant enhancement to the project's code quality by implementing the `typos` tool. It serves to identify and correct spelling errors throughout the codebase. Additionally, a `typos CI workflow` has been configured to ensure ongoing spelling accuracy in future contributions.

## Key Changes

1. **Spelling Corrections**: All identified spelling mistakes have been corrected across various files in the project, improving the readability and professionalism of the code.

2. **Typos CI Workflow**: The integration of `typos` into our continuous integration workflow adds an automatic spell check step. This ensures that new commits adhere to the same level of spelling accuracy, maintaining the overall quality of our code.

3. **Documentation Update**: Where applicable, documentation has also been updated to reflect the spelling corrections.

### CODE REVIEW

File: .github/workflows/typos.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: Spell Check 
+on:
+  pull_request:
+  push:
+    branches:
+      - "main"
```
Comment (nmn) [inline .github/workflows/typos.yml:6]: Wouldn't we want it to run on PRs and not the `main` branch?

**Review by nmn — COMMENTED** [review]
This looks really good. I just want to make sure it's configured to run on PRs.

File: .github/workflows/typos.yml (hunk 1 above)
Comment (liby) [reply .github/workflows/typos.yml:6]: I can confirm that the workflow is indeed configured to run on pull requests. The `pull_request` trigger in our workflow file ensures that the spell check will run for every new or updated pull request, in addition to the `push` events on the main branch. 

Please let me know if there are any other aspects you would like me to address or if everything looks good to go.

Comment (nmn) [reply .github/workflows/typos.yml:6]: Thanks! This is a great improvement.

**Pull request merged**